crates/relayer-framework-test/src/relayer_mock/util/mutex.rs

```diff
@@ -0,0 +1,19 @@
+//! MutexUtil is a helper trait to simplify the error handling
+//! when locking a Mutex.
+
+use std::sync::{Arc, Mutex, MutexGuard};
+
+use crate::relayer_mock::base::error::Error;
+
+pub trait MutexUtil<T> {
+    fn acquire_mutex(&self) -> Result<MutexGuard<T>, Error>;
```

Comment:
Sorry for not being clear, I think we can have a non-fallible version of `acquire_mutex` here. For this specific case, we can treat poisoned mutex errors as fatal errors during tests that they can just panic. We can add an `#[allow()]` pragma here for an exception for allowing panic.

With that, the functions that only use the mutex can return the result straight without needing to return `Result`.